### ISSUE TITLE
Multiple VestaCP-niiknow Instances on one VPS with one IP

### ISSUE THREAD

**Issue by michacassola** [niiknow/vestacp#8]
Hi,

how would I solve this issue of having multiple seperate instances of dockerized vesta on one VPS?
I want to limit RAM and CPU through Docker for each user who will get one of the installation.

Best
Micha

**Comment by noogen** [comment]
To do something like that, you can try looking at [dokku](https://github.com/dokku/dokku); but you should build your own Vesta docker image.  My image just have too much stuff in there to be efficient for dokku.  You almost don't need VestaCP at that point.

Due to IP and ports limitation, the best way to do it is at the Vesta code level. But unlike cPanel, VestaCP currently does not have this kind of support natively.  It is also out of the scope of this project.

**Comment by michacassola** [comment]
Thank you very much!

What stuff should I leave out for dokku so my image would be efficient?

What do you mean by: "You almost don't need VestaCP at that point."?
Does dokku have a panel that I can use to easily make RAM/CPU/Bandwidth-separated Wordpress Instances?

The part about IP/Port limitations I do not understand.

**Comment by noogen** [comment]
> What stuff should I leave out for dokku so my image would be efficient?

A lot of it.  Pretty much just bare VestaCP.

> What do you mean by: "You almost don't need VestaCP at that point."?

See my explanation in IP/Port limitations.

> Does dokku have a panel that I can use to easily make RAM/CPU/Bandwidth-separated Wordpress Instances?

dokku basically proxy the front-end with nginx to your backend docker.   All your RAM/CPU is set in docker https://docs.docker.com/engine/admin/resource_constraints/.  Bandwidth is set in VestaCP.

> IP/Port limitations.

VestaCP is not only a webserver.  You also have DNS, email, and all sort of other things that requires specific port to be available.  Your single IP limitation would become an issue, especially more complicated when SSL is involved.

At this point, it's either VestaCP does it natively in code or you implement your own proxy docker base on host header.  Example: https://github.com/dlundquist/sniproxy

And if you're doing the sniproxy route, You might as well drop Vesta since dokku already have strategy to create PaaS setup like you want.  You have to read up more on doing it the dokku way.

Basically, it is a lot of work to either use (dokku + custom proxy) or wait for Vesta to implement it.  You're better off using cPanel as it already does what you want.

**Comment by noogen** [comment]
What I'm saying is that the route you want to do will actually result in a  lot more maintenance and headache when additional VPS is actually pretty cheap @ only $5/month: https://www.linode.com/pricing

**Comment by michacassola** [comment]
Thank you again for you great answers!

I found a host where they give you an unbeliavable amount of resources for not too much but also not so little that it will fit for my smaller plans. Their smallest VPS is already huge.
And I thought a containerized layer will also maybe give more security...

But you are right I should keep it more simple.
Maybe I'll look into dokku more.
